source/core_json.c

```diff
@@ -256,69 +267,62 @@ static bool_ skipUTF8( const char * buf,
  *
  * @param[in] c  The character to convert.
  *
- * @return the integer value upon success or UINT8_MAX on failure.
+ * @return the integer value upon success or NOT_A_HEX_CHAR on failure.
  */
+#define NOT_A_HEX_CHAR    ( 0x10U )
```

Comment:
These are static functions.  This doc is for the programmer reading the source, and not for general exposure (and wrapped in a DO_NOT_DOCUMENT block).  I prefer one-off macros to live close to their first use, which makes it easier to read the code in a linear fashion.

Reply:
If this doesn't result in doxygen errors, I'm fine with that then. Though, I think this does go against our style guidelines of declaring macros at the top, so we should update that if not done already in https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1205

Reply:
We only have global variables at the top of the file, and local variables at the top of a block. There's nothing we have against not having macros at the top.

Reply:
I am generally happy with declaring macros close to where they are needed, but in this case you are loosing the context that makes 0x10U not a magic number.  Defining a "magic" number 23 lines before it is used does not make it not-magic.  In this case why 0x10 instead of 0x20?

Reply:
It's defined right after the comment which describes it as the return value for an error.  NOT_A_HEX_CHAR is just what it says it is.  There’s no magic to it.  Previously I used UINT8_MAX, but coverity complained because the value (255) was signed.  Any value that does not correspond to a legal hex char would do.  Had this been signed, I would have picked -1.